### PR TITLE
feature(global-cta-modal-implementation): Global CTA Video Modal.

### PR DESCRIPTION
**Pull request by nimithshetty17**
## Issue

Fixes #[MERATIVE-858](https://jira.sdlc.merative.com/browse/MERATIVE-858)

## Description

We need to ensure that all of the button styles (e.g.. Primary, Secondary, Tertiary) (see - https://main--merative2--hlxsites.hlx.page/block-library/buttons/buttons-video) can support video modal when the href is defined as a video type.

**Changed**

image

- Figma Link - https://www.figma.com/file/vw24IPXXt4vCh9BNumka9A/Web--Merative-Digital-Design-System-2.0?type=design&node-id=272-62&mode=design&t=AV0llNTB4XL0rh71-0

## Test URLs
  
- Before (Changes from `main`): https://main--merative2--nimithshetty17.hlx.page/block-library/buttons/buttons-all
- After (Changes from this PR): https://global-cta-modal-implementation--merative2--nimithshetty17.hlx.page/block-library/buttons/buttons-all
  
## Testing Instruction

To test if the video player is opening a modal for all the button and link styles.

### CODE REVIEW

**Review by sharathmrft — APPROVED** [review]
@nimithshetty17 
The video player is opening a modal for all the button and link styles.

@proeung @sachinmesh

**Review by sachinmesh — APPROVED** [review]
All CTAs are opening the modal.

File: blocks/linkable-cards/linkable-cards.js [hunk 1]
```diff
@@ -54,6 +56,20 @@ export default async function decorate(block) {
...
+      const videoModal = buildVideoModal(a.href, videoType);
+      const videoClose = videoModal.querySelector('button.video-modal-close');
+      videoClose.addEventListener('click', () => toggleVideoOverlay(videoModal));
+      a.addEventListener('click', (e) => {
+        e.preventDefault();
+        toggleVideoOverlay(videoModal);
+      });
+      block.appendChild(videoModal);
+    }
+  });
```
Comment (proeung) [inline blocks/linkable-cards/linkable-cards.js:71]: Nitpicking: Let's add comments and adjust the format to make the code more structured.

```suggestion
// Select all anchor elements within the 'block' container
block.querySelectorAll('a').forEach((a) => {
  // Determine the type of video based on the href attribute
  const videoType = getVideoType(a.href);

  // Check if the video type is 'youtube' or 'mp4'
  if (['youtube', 'mp4'].includes(videoType)) {
    // Build a modal for the video
    const videoModal = buildVideoModal(a.href, videoType);

    const videoClose = videoModal.querySelector('button.video-modal-close');

    // Add a click event listener to close the video modal when the close button is clicked
    videoClose.addEventListener('click', () => toggleVideoOverlay(videoModal));

    a.addEventListener('click', (e) => {
      e.preventDefault();

      // Toggle the video overlay when the anchor element is clicked
      toggleVideoOverlay(videoModal);
    });

    // Append the video modal to the 'block' container
    block.appendChild(videoModal);
  }
});
```

File: scripts/lib-franklin.js [hunk 2]
```diff
@@ -534,6 +703,17 @@ export function getIconTypebyPath(url) {
...
+    if (['youtube', 'mp4'].includes(videoType)) {
+      const videoModal = buildVideoModal(a.href, videoType);
+      const videoClose = videoModal.querySelector('button.video-modal-close');
+      videoClose.addEventListener('click', () => toggleVideoOverlay(videoModal));
+      a.addEventListener('click', (e) => {
+        e.preventDefault();
+        toggleVideoOverlay(videoModal);
+      });
+      element.appendChild(videoModal);
+    }
```
Comment (proeung) [inline scripts/lib-franklin.js:716]: Let's add comments and format this similar to the suggestion above.

**Review by proeung — CHANGES_REQUESTED** [review]
@nimithshetty17 Looks good overall! I just have minor code suggestion, but the modal overlay is rendering when the a link is set to YT or MP4 paths. https://global-cta-modal-implementation--merative2--nimithshetty17.hlx.page/block-library/buttons/buttons-all

Screenshot 2023-09-13 at 2 16 49 PM

File: scripts/lib-franklin.js (hunk 2 above)
Comment (nimithshetty17) [reply scripts/lib-franklin.js:716]: @proeung for the first point, could you please explain a bit on that overlay issue?

**Review by proeung — APPROVED** [review]
LGTM! Thanks for addressing my code suggestions.

**Comment by proeung** [comment]
@nimithshetty17 I decided to revert this PR from the `main` branch due to some of the 404 errors in the console log (see attached). Looks like the video model gets added to every block element (https://github.com/hlxsites/merative2/pull/335/files#diff-a5ee87ff9063f921ba01933977e6164e5729551986030f0ed7eb747bcffb740fR727), which tells the the `loadBlock` function to build and load in the blocks (https://github.com/hlxsites/merative2/blob/main/scripts/lib-franklin.js#L349) for classes `.video-modal-overlay` and `.video-modal-container`.

You might need to adjust your approach to appending the video overlay outside of the `main` HTML tag and append the video embed onClick.

Screenshot 2023-09-15 at 1 06 41 PM

Let's open a new PR for this work and include that fix that eliminates the 404 error in the console log.

**Pull request merged**